modules/cats/build.sbt

```diff
@@ -2,7 +2,8 @@ name := "pureconfig-cats"
 
 libraryDependencies ++= Seq(
   "org.typelevel" %% "cats-core" % "1.4.0",
-  "org.typelevel" %% "cats-laws" % "1.4.0" % "test")
+  "org.typelevel" %% "cats-laws" % "1.4.0" % "test",
+  Dependencies.shapeless % Provided)
```

Comment:
Hi @ruippeixotog , The ConfigReader/Writer I defined depends on `Shapeless` to prevent it from overriding the implicits already defined for `TraversableOnce` and `Option`. I could make it less generic and only work for `Chain`, or I could define a mini `Shapeless` like this:
```scala
// Derived from Shapeless (https://github.com/milessabin/shapeless)
object MiniShapeless {
  def unexpected: Nothing = sys.error("Unexpected invocation")

  type ¬[T] = T => Nothing
  type ¬¬[T] = ¬[¬[T]]
  type ∧[T, U] = T with U
  type ∨[T, U] = ¬[¬[T] ∧ ¬[U]]

  @scala.annotation.implicitNotFound("${A} must not be a subtype of ${B}")
  trait <:!<[A, B] extends Serializable

  implicit def nsub[A, B]: A <:!< B = new <:!<[A, B] {}
  implicit def nsubAmbig1[A, B >: A]: A <:!< B = unexpected
  implicit def nsubAmbig2[A, B >: A]: A <:!< B = unexpected
}
```

Reply:
Hmm, I see what you mean. I strongly want to avoid a shapeless dependency here just for this use case, but an alternative solution to defining a `MiniShapeless` would be to use our mechanism to export low-priority readers. If you define an implicit instance of `Exported[ConfigReader[A]]` instead of a `ConfigReader[A]`, that instance will still be used when a `ConfigReader[A]` is requested but only if an instance of `ConfigReader[A]` is not found.

I'd like not to abuse `Exported` in our codebase, but this seems like a valid use case for `Exported`, right @jcazevedo and/or @leifwickland?

Reply:
Will have a look at `Exported`.

Reply:
@ruippeixotog All change request resolved unless there are objections to the use of `Exported`.